### PR TITLE
[DF] Test lazy instant actions in distributed mode

### PR DESCRIPTION
**Pull request by vepadulano**
Sibling PR of https://github.com/root-project/root/pull/9995

### CODE REVIEW

File: python/distrdf/dask/test_reducer_merge.py [hunk 1]
```diff
@@ -252,6 +254,34 @@ def test_distributed_asnumpy_columns(self):
         int_32_dtype = numpy.dtype("int32")
         self.assertEqual(npy_x.dtype, int_32_dtype)
 
+    def test_distributed_asnumpy_lazy(self):
+        """Test that `AsNumpy` can be still called lazily in distributed mode"""
```
Comment (etejedor) [inline python/distrdf/dask/test_reducer_merge.py:258]: Can we have a common helper function for `test_distributed_asnumpy` and `test_distributed_asnumpy` that accepts a `lazy` parameter? 90 % of the code of that helper function would be the same no matter the value of `lazy`.

File: python/distrdf/dask/test_reducer_merge.py [hunk 2]
```diff
@@ -308,6 +338,36 @@ def test_distributed_snapshot_columnlist(self):
         for filename in tmp_files:
             os.remove(filename)
 
+    def test_distributed_snapshot_lazy(self):
+        """Test that `Snapshot` can be still called lazily in distributed mode"""
+        # A simple dataframe with ten sequential numbers from 0 to 9
+        df = Dask.RDataFrame(10, daskclient=self.client)
+
```
Comment (etejedor) [inline python/distrdf/dask/test_reducer_merge.py:345]: Same comment as above for `test_distributed_snapshot` and `test_distributed_snapshot_lazy`.

File: python/distrdf/spark/test_reducer_merge.py [hunk 3]
```diff
@@ -266,6 +269,34 @@ def test_distributed_asnumpy_columns(self):
         int_32_dtype = numpy.dtype("int32")
         self.assertEqual(npy_x.dtype, int_32_dtype)
 
+    def test_distributed_asnumpy_lazy(self):
+        """Test that `AsNumpy` can be still called lazily in distributed mode"""
+
```
Comment (etejedor) [inline python/distrdf/spark/test_reducer_merge.py:274]: Same comment as above.

File: python/distrdf/spark/test_reducer_merge.py [hunk 4]
```diff
@@ -325,6 +356,36 @@ def test_distributed_snapshot_columnlist(self):
         for filename in tmp_files:
             os.remove(filename)
 
+    def test_distributed_snapshot_lazy(self):
+        """Test that `Snapshot` can be still called lazily in distributed mode"""
+        # A simple dataframe with ten sequential numbers from 0 to 9
```
Comment (etejedor) [inline python/distrdf/spark/test_reducer_merge.py:361]: Same comment as above.

**Comment by vepadulano** [comment]
@phsft-bot build

**Pull request merged**